WordPress/gutenberg
# Query Loop: Default to querying posts when on singular content

## What?

An attempt to try and fix the issue described in https://github.com/WordPress/gutenberg/issues/64746.

## Why?

To align the default/inherit query results between the Editor and the front end, and also hopefully make this block more intuitive and easier to understand.

## How?

This adds a check for `is_singular()` to check if the block has been added to a single post or page (i.e. within content), and if so, defaults to querying the `post` post type.

It also hides the Query Type control when editing a single page/post/CPT, as these blocks will always default to a custom query type that queries posts. The Query Type control is still used when editing a template, where it makes more sense to default to inheriting the global query.

## Testing Instructions




1. Insert a Query Loop block inside a single page, post, or CPT.
2. Ensure that the Default query ("inherit") shows posts from the 'post' post type in both the Editor and the front-end.
3. Insert a Query Loop block inside a template.
4. Ensure that the "Query Type" control is only shown when editing a template and that it is hidden when editing a single page, post, or CPT.

## Screenshots or screencast 

### Front-end (posts should now render by default)

| Before | After |
| ----- | ----- |
| <img width="690" alt="image" src="https://github.com/user-attachments/assets/99e88f9b-7334-41eb-ada3-a4562e1c6356"> | <img width="690" alt="image" src="https://github.com/user-attachments/assets/921656c9-3d95-412a-93b7-0ac4e8887219"> |

### Editor (shouldn't be any change)

| Before | After |
| ----- | ----- |
| <img width="1433" alt="image" src="https://github.com/user-attachments/assets/df875a12-9c37-48ff-8cdd-55a8f5d10566"> | <img width="1433" alt="image" src="https://github.com/user-attachments/assets/f9aefa1b-0ab4-4b6f-9f32-aba34b1efdcb"> |

### Query Type control changes (currently, the Query Type control is always shown)

| Editing a template | Editing content (e.g. page) |
| -------- | ------- |
| <img width="675" alt="image" src="https://github.com/user-attachments/assets/0749b362-3f1b-476f-b749-c59fcb605b47"> | <img width="675" alt="image" src="https://github.com/user-attachments/assets/1ae1666c-b0e4-4412-b73f-964f3ba999fb"> |
